### ISSUE TITLE
Midnight Club: Los Angeles stuck on intro. Regression?

### ISSUE THREAD

**Issue by lextra2** [RPCS3/rpcs3#6741]
CPU: Ryzen 5 3600
GPU: RX 580 4GB - Driver: 19.9.2
OS: Windows 10 1903
RPCS3 Build: 0.0.7-8834

Midnight Club: Los Angeles [BLES00652] gets stuck immediately after launching.

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/3715128/RPCS3.log)

**Comment by dio-gh** [comment]
Needs regression testing done. See [here](https://wiki.rpcs3.net/index.php?title=Help:Using_different_versions_of_RPCS3#How_to_find_the_build_that_caused_a_regression.3F) for details. Basically, we'd need which PR broke it.

**Comment by lextra2** [comment]
I will try to nail it down.

**Comment by lextra2** [comment]
Ok so build 8363 works while 8364 does not. The change in the newer build of the two is this one #6193
which makes LLVM Recompiler the default.

Conclusion: Using the ASMJIT Recompiler (the old default) on the build (8834) allows for going ingame once again.

**Comment by dio-gh** [comment]
That PR cannot have broken it, since it's just cosmetic. You're basically saying that you can't get the game to work with SPU LLVM. That's indeed an issue, but not unexpected - MCLA is Ingame, instability is expected.

So I'm not really sure what's the issue here? If SPU LLVM broke, test where that broke. If SPU LLVM never worked for you, or it was always unreliable to get working, that's normal considering the game's compat status, and this ticket should be then closed.

**Comment by lextra2** [comment]
Ok gonna find out what build broke SPU LLVM for this game (if it ever worked to begin with)

**Comment by dio-gh** [comment]
I'm fairly sure no regressions are here actually, you're just being unlucky with some race conditions. Try altering some settings (e.g. modifying the preferred spu threadcount, changing the blocksize, etc.) to get the timings different a bit. To me this seems like more of a support issue that should be discussed on Discord, really.

**Comment by lextra2** [comment]
The reason why assumed regression is because it didn't go ingame with the default settings. I'm still investigating what on what build LLVM stopped going ingame.

**Comment by lextra2** [comment]
Alright. You were correct. Adjusting `SPU Block Size` from `Safe` to `Mega` allowed me to go ingame just fine.

**Comment by dio-gh** [comment]
That points to general instability then. You should close this issue.

You're just not winning the particular race condition, this really doesn't seem to be an issue with the `Safe` blocksize.

**Comment by Asinin3** [comment]
Yes this game is just unstable with SPU LLVM it doesn't matter what blocksize you use. Me and Yahfz were trying to get it running for about 4 hours back in august for a youtube video. Basically i found that it was more stable on TSX cpu's which is just the way things are in many cases on RPCS3 right now. It doesn't seem like a very specific issue with the game and just general race conditions with SPU LLVM / SPU LLVM exacerbating the problem. Can be closed imo.
